File: backend_api/models/Book.js
const mongoose = require("mongoose");

const productSchema = new mongoose.Schema({
  title: { type: String, required: true },
  description: { type: String },
  price: { type: Number, required: true },
  image_url: { type: String, required: true },
  category: { type: String, required: true },
  publisher: { type: String },
  year: { type: Number },
});

// The third argument specifies the exact collection name 'product' in your MongoDB.
module.exports = mongoose.model("Product", productSchema, "product");

File: app/src/main/java/vn/hcmute/ktqt/models/Book.java
package vn.hcmute.ktqt.models;

import com.google.gson.annotations.SerializedName;

public class Book {
    @SerializedName("_id")
    public String id;

    public String title;
    public String description;
    public double price;

    @SerializedName("image_url")
    public String imageUrl;

    public String category;
    public String publisher;
    public int year;
}
